### PR TITLE
[22884] adds USiP logging to SiS auth & refactors applications constants

### PR DESCRIPTION
**Pull request by bramleyjl**
## Summary

- This PR removes the standalone Universal Sign-in Page logging that existed for SSOe `/new` & `/callback` methods; replacing them with updates to all SSOe `/new` & `/callback` logs to include `application` URL parameter that USiP depends on. The following StatsD logs and their corresponding Rails logs:
  * 'api.auth.new' / `SSO_NEW_KEY`
  * 'api.auth.login.success' / `LOGIN_STATUS_SUCCESS`
  * 'api.auth.login.failure' / `LOGIN_STATUS_FAILURE`
will now contain a `client_id:<application>` tag that will be populated if the login originated through the USiP. This 
- Given that the USiP either the SiS or SSOe flow and that both of those flows have extensive logging, it made sense to remove an duplicative logging solution and expand what is already used and maintained.
- USiP and SSOe are both maintained by Identity

## Related issue(s)
- https://github.com/department-of-veterans-affairs/vets-website/issues/22884
- [*Original USiP logging PR*](https://github.com/department-of-veterans-affairs/vets-api/commit/5fd1d2333ca53d6fd488583db6b120b77371b86d)

## Testing done

- *Describe what the old behavior was prior to the change*: USiP logins would trigger the 'api.auth.unified_new' & 'api.auth.unified_callback' StatsD logs
- *Describe the steps required to verify your changes are working as expected*:
  * Perform an SSOe login with any of the USiP clients: `http://localhost:3001/sign-in/?application=ebenefits`
  * Look for SSO_NEW_KEY & LOGIN_STATUS_SUCCESS tags that include your chosen USiP application as the `client_id`. You can also artificially throw an error in the callback process and look for the corresponding LOGIN_STATUS_FAILURE Rails log.
- *Describe the tests completed and the results*: Unit specs that cover this area of SSOe logging have been updated to include `client_id`.

## Logs
* `2022-12-12 10:06:15.276810 I [20463:puma srv tp 015] Rails -- SSO_NEW_KEY, tags: ["context:idme", "version:v1", "client_id:ebenefits"]`
* `2022-12-12 10:06:45.640282 I [20463:puma srv tp 002] Rails -- LOGIN_STATUS_SUCCESS, tags: ["context:idme", "version:v1", "client_id:ebenefits"]`
* `2022-12-12 10:11:26.242487 I [23470:puma srv tp 007] Rails -- LOGIN_STATUS_FAILURE, tags: ["context:idme", "version:v1", "client_id:mhv", "error:007"], message: undefined method '[]' for nil:NilClass`

## What areas of the site does it impact?
This PR impacts the SSOe login flow, all SSOe logins will now include the `client_id` tag in the `/new` & `/callback` phases of the authentication flow. No functionality is impacted as this is purely a logging change.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by holdenhinkle — DISMISSED** [review]
LGTM 🤘

**Comment by bosawt** [comment]
Tested by authenticating with `application=ebenefits`, confirmed the logs:
`SSO_NEW_KEY, tags: ["context:idme", "version:v1", "client_id:ebenefits"]`
`LOGIN_STATUS_SUCCESS, tags: ["context:idme", "version:v1", "client_id:ebenefits"]`

Also tested by authenticating with `application=nothing`, confirmed the logs:
`SSO_NEW_KEY, tags: ["context:idme", "version:v1", "client_id:nothing"]`
` LOGIN_STATUS_SUCCESS, tags: ["context:idme", "version:v1", "client_id:nothing"]`

**Comment by bosawt** [comment]
The va.gov USIP calls backend without an application param, right? Wonder if we could set a default for this when `application` isn't defined, like `vaweb`. That way we'd be able to explicitly know the web calls, instead of assuming it happens when `client_id` is blank

**Review by bosawt — CHANGES_REQUESTED** [review]
Just had a small comment about setting a default, otherwise all looks good! 

I put in a separate comment but I tested as well and looks good

File: lib/saml/url_service.rb [hunk 1]
```diff
@@ -21,8 +21,8 @@ class URLService
 
     LOGIN_REDIRECT_PARTIAL = '/auth/login/callback'
     LOGOUT_REDIRECT_PARTIAL = '/logout/'
-    STATSD_SSO_UNIFIED_NEW_KEY = 'api.auth.unified_new'
     BROKER_CODE = 'iam'
+    UNIFIED_SIGN_IN_CLIENTS = %w[mhv myvahealth ebenefits vamobile vaoccmobile].freeze
```
Comment (bosawt) [inline lib/saml/url_service.rb:25]: We also need to add `vaweb` here now, right?

**Review by bosawt — APPROVED** [review]
looks good!

**Review by bosawt — APPROVED** [review]
👍

**Pull request merged**